app/models/supplier.rb

```diff
@@ -51,8 +53,9 @@ def sync_all
     end
     # Find any new articles, unless the import is manual
     unless shared_sync_method == 'import'
-      for shared_article in shared_supplier.shared_articles
-        unless articles.undeleted.find_by_order_number(shared_article.number)
+      # for shared_article in shared_supplier.shared_articles
+      for shared_article in shared_supplier.shared_articles.all
```

Comment:
could we filter the existing articles already on the database level with something like `shared_supplier.shared_articles.where(available: true).where.not(id: existing_articles)`?

Reply:
i care more about the readability than the performance.
thanks for your patience and i'm looking forward to your next PR!